### ISSUE TITLE
[Discovery] Improve location/facility selection experience

### ISSUE THREAD

**Issue by lenaecb** [department-of-veterans-affairs/va.gov-team#12549]
Facility selection in the scheduling flow for VA appointments is tripping up users. Let's make it better. 

## User problem: 
When Veteran is going through VAOS scheduling flow for VA requests, when they reach "Select a location" page they do not immediately recognize the options we are presenting them. These options at are the parent site level and do not include city/state. 

From user testing, Veterans did not immediately know what to do or what to select if they didn't recognize the names. 
- Every Veteran participant that interacted with VAOSR (12; 80%) had more than one VAMC affiliated with their account and 100% of these Veterans struggled to find the location where they wanted to schedule an appointment.

We have observed dropoff on the Facility selection page that can now be attribute to both limited eligibility and UX based on user feedback. 
![Screen Shot 2020-08-18 at 2.41.50 PM.png](https://images.zenhubusercontent.com/5e41a7217d1a39827f6e83fe/ef2748c9-ac83-419a-9db9-7c75dd191430)

## Hypothesis:
We believe that by improving the facility selection experience we will improve:
- Conversation rates
- Decrease dropoff rates on facilities page
- Improve user satisfaction

**Comment by rtwell** [comment]
OK gang—here's a couple comps demonstrating the location based sort for facilities. 

Some high level design decisions:

1. We flattened the parent/child facility hierarchy, and instead display _all_ facilities a Veteran can request an appointment. 
2. We will pair the *facility name* with its respective *city, state*.

This approach introduces a new problem: sometimes the list of facilities will be long, making it difficult to parse the list of facilities. Given that, we could tackle this issue with two alternatives:
1. Sort the list by location.
2. Sort the list alphabetically
3. Regardless of sort, we could decide to display the "top 5" facilities to Veterans and if they don't see a facility they want to select, they can expand the list and see the rest. 

Sorting by location would need an affordance like milage to indicate how it's sorted, as it's not apparent immediately. Also, users could change the origin location (which will default to the residential address on file) to use their current location by clicking the link and allowing the browser to use the device GPS and sort from there. If no address on file, we can default to alpha order, but still provide  the current location option.

Something to note on the alphabetized list—when a Veteran lives close to state lines, it gets a little wonky as the locations can vary wildly (see the far right example).

### From left to right: Sort by location default, Sort by location expanded, Sort alphabetically

![image](https://user-images.githubusercontent.com/12564977/91559568-58604080-e906-11ea-9b89-2762aa2bcf11.png)

**Comment by lenaecb** [comment]
Thanks, @rtwell 

**A few questions:**
- @rtwell - For Veteran's address: What if Veteran wants to use an address that is neither their address on file or their current location? I am wondering also if folks will get tripped up on how to use the 'use my current location.' I know in other areas of the app we allow for personal information to be updated in a field (like email, phone). Would that actually add more confusion?
- @jbalboni @narin - we don't yet if can use miles (instead of minutes) for our sorting affordance, correct?

**Comments:**
- I think sorting based on location proximity to residential city, state is more intuitive
- I want to add some analytics on this page so we can actually get a sense of avg # of facilities a Veteran is registered to so we have a better idea of how long some lists can get _(I can try and find a rough estimate out in the VA data world)_
- I like the idea of showing top 5 or 10, but I noticed in our testing that sometimes the nesting gets overlooked... it might be worth trying to show all and getting a sense of how many facilities that actually is for a given user. Thoughts?

**Comment by rtwell** [comment]
@lenaecb 

> What if Veteran wants to use an address that is neither their address on file or their current location?

the goal is to surface facilities in an intuitive (enough) way to help folks make a selection quickly. no matter which approach we take, some segment of folks' facilities will end up at the bottom of the list (because the location they want is far away, or the name is towards the end of the alphabet). both of those scenarios are far less frustrating for users that aren't registered at many facilities, because they will simply see a shorter list, and that's when the sorting doesn't _really_ matter because users can see all of the options at a glance. I would like to test this with Veterans to see if that's a feature they want/need to be met—I do not want this to turn into a facility locator tool. I love your suggestion of adding analytics for this page b/c we can learn / have data about how many facilities end up getting displayed.

**Comment by lenaecb** [comment]
We definitely will test, but I do want to move to implementing a next iteration and use our feature toggle to display this new selection experience to some % of Veterans along with manual testing.

I am still hunting for this data - avg # of facilities a Veteran is registered

Nonetheless, outside of wanting the data point above, sounds like we're at a decision point for next iteration to test with Veterans. I am advocating for:
- using distance (miles) rather than drive time (minutes) + sorting based on location proximity to residential city, state
- launching this next iteration in phases to measure if dropoff rate + do some manual testing with users
- add analytics to this page to get facilities on avg are displaying for a user

@lalexanderson-dsva - feedback on the above? (both ryan's mocks/thoughts + my recommenation above)

**Comment by jbalboni** [comment]
@lenaecb To answer your question above, we definitely can use straight line distance in miles for the sorting and I would also agree that that's what we should use, since facility locator does the same thing.

**Comment by lenaecb** [comment]
Data point to note: based on input from folks on the identity team/MVI, they estimate that Veterans will see 5 or less with known outliers potentially having up to 25 facilities. 

I think we use this as our hypothesis and add analytics to see how close or far off that hypo is. So, I would say we could also filter by 5 at first and see on avg how many folks actually have.

**Comment by jbalboni** [comment]
@lenaecb I think that's 5 VistA sites right? That would be translate to a lot more actual facilities in our list.

**Comment by lenaecb** [comment]
Yep, @jbalboni you're correct... revising my statement above. The estimate is that Veterans are registered at **5 VistA sites** with outliers being somewhere upwards to 25 sites :grimacing:

**Comment by peggygannon** [comment]
Hello, @lenaecb and @rtwell Here is content recommendation for this screen. Please let me know if you have any questions. 

**Choose a VA location for your appointment**
Below is a list of VA locations where you’re registered that offer [typeofcare] appointments. Locations closest to you are at the top of the list. We base this on the address we have on file for you. Please select where you’d like to have your appointment

**Comment by rtwell** [comment]
@peggygannon what do you think about adding the type of care to the heading? For example, **Choose a VA location for your Podiatry appointment** Would that get too wordy? My reasoning is i feel like it could be a useful piece of info for Veterans and it could be overlooked.

**Comment by rtwell** [comment]
also, @peggygannon are you OK with the rest of the language on the page, specifically "Locations close to your address we have on file:" idk it feels clunky to me, but i've been staring at this too long and do not trust myself.

**Comment by peggygannon** [comment]
Hi, @rtwell 

Yes, my concern with adding {typeofcare} to the header is that it would get wordy and bump the header down to 3 lines on mobile. I do agree it's an important piece of info though. Do you think bolding it would help (see below)? 

also, the address on file info got cut off when I uploaded my recommendation earlier. here it is below. 

-------

H2?**Choose a VA location for your appointment**
Below is a list of VA locations where you’re registered that offer [typeofcare] appointments. Locations closest to you are at the top of the list. We base this on the address we have on file for you. Please select where you’d like to have your **[type of care]** appointment.

H3? **Your address on file:** 
{address line 1 on file}
{address line 2 on file}

Or, [use your current location]

**Comment by rtwell** [comment]
hmmm yeah i think i'm ok with a bit of scrolling if it means we're providing more clarity. What's the longest type of care we offer? I think it's "MOVE! weight management program". I'll drop that in and see how it looks.

**Comment by rtwell** [comment]
I'm OK with the ToC in the header, I think. Any objections?

![image](https://user-images.githubusercontent.com/12564977/92120290-d9b44900-edc6-11ea-8ea4-45243f9d5e66.png)

**Comment by rtwell** [comment]
Oh, also @peggygannon it just occurred to me we need text for the fallback (alphabetical) version. I think I can piece together a first pass, but would love to get your eyes on it. We need a fallback version in case we don't have an address on file for a user:

![image](https://user-images.githubusercontent.com/12564977/92120948-afaf5680-edc7-11ea-99cd-9e855f6292f2.png)

**Comment by jbalboni** [comment]
Yeah, we would filter out any facilities that are not configured to support your type of care in VATS.

There will still be the additional checks (request limit, past visits, available clinics) that happen after you choose one of the options.

**Comment by peggygannon** [comment]
@rtwell looks good. Thank you!
